internal/api/controllers/private/highlevelConnectionStatus.go

```diff
@@ -0,0 +1,240 @@
+package private
+
+import (
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/api/connectors"
+	"playbook-dispatcher/internal/api/connectors/inventory"
+	"playbook-dispatcher/internal/api/connectors/sources"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/labstack/echo/v4"
+)
+
+type rhcSatellite struct {
+	SatelliteInstanceID      string
+	SatelliteOrgID           string
+	SatelliteVersion         string
+	Hosts                    []string
+	SourceID                 string
+	RhcClientID              *string
+	SourceAvailabilityStatus *string
+}
+
+func (this *controllers) ApiInternalHighlevelConnectionStatus(ctx echo.Context) error {
+	var input HostsWithOrgId
+	satelliteResponses := []RecipientWithConnectionInfo{}
+	directConnectedResponses := []RecipientWithConnectionInfo{}
+	noRHCResponses := []RecipientWithConnectionInfo{}
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	hostConnectorDetails, err := this.inventoryConnectorClient.GetHostConnectionDetails(
+		ctx.Request().Context(),
+		input.Hosts,
+		this.config.GetString("inventory.connector.ordered.how"),
+		this.config.GetString("inventory.connector.ordered.by"),
+		this.config.GetInt("inventory.connector.limit"),
+		this.config.GetInt("inventory.connector.offset"),
+	)
+
+	if err != nil {
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	satellite, directConnected, noRhc := sortHostsByRecipient(hostConnectorDetails)
+
+	// Return noRHC If no Satellite or Direct Connected hosts exist
+	if noRhc != nil {
+		noRHCResponses = []RecipientWithConnectionInfo{getRHCStatus(noRhc, input.OrgId)}
+	}
+
+	if satellite == nil && directConnected == nil {
+		return ctx.JSON(http.StatusAccepted, noRHCResponses)
+	}
+
+	if len(satellite) > 0 {
+		satelliteResponses, err = getSatelliteStatus(ctx, this.cloudConnectorClient, this.sourcesConnectorClient, input.OrgId, satellite)
+
+		if err != nil {
+			return fmt.Errorf("Error retrieving Satellite status: %s", err)
```

Comment:
Should you return an error here?  Or should you log the error (and the request_id) and allow the processing to proceed and return a chunk of valid information to the caller?